### ISSUE TITLE
saved model: Expected image, got empty file

### ISSUE THREAD

**Issue by baracoder** [NVIDIA/triton-inference-server#1454]
**Description**

I might be missing something, but everything I tried so far, was not successful. 
Test model, which works on tensorflow-serving, fails with triton:

The result is
```
request_status {
  code: INTERNAL
  msg: "Expected image (JPEG, PNG, or GIF), got empty file\n\t [[{{node map/while/decode_jpeg}}]]"
  server_id: "inference:0"
  request_id: 797
}
meta_data {
}
```

saved model signature according to `saved_model_cli show` is
```
signature_def['serving_default']:
  The given SavedModel SignatureDef contains the following input(s):
    inputs['JpegBatch'] tensor_info:
        dtype: DT_STRING
        shape: (-1)
        name: JpegBatch:0
  The given SavedModel SignatureDef contains the following output(s):
    outputs['Activation'] tensor_info:
        dtype: DT_FLOAT
        shape: (-1, -1)
        name: tinyaverager/embedding:0
  Method name is: tensorflow/serving/predict
```
the first `-1` should be the batch.

The triton generated config is
```
name: "tiny_averager"
platform: "tensorflow_savedmodel"
max_batch_size: 1
input {
  name: "JpegBatch"
  data_type: TYPE_STRING
  dims: 1
  reshape {
  }
}
output {
  name: "Activation"
  data_type: TYPE_FP32
  dims: -1
}
```

Might be related to https://github.com/NVIDIA/triton-inference-server/issues/1328 but I am not sure.

**Triton Information**

nvcr.io/nvidia/tritonserver:20.03-py3

**To Reproduce**

saved model: https://storage.googleapis.com/static.nyris.io/bug-reports/triton/tfserving_tiny_averager_1_model.savedmodel_saved_model.pb

Python code for interface using grpc:
```
    import grpc
    from tensorrtserver.api import grpc_service_pb2, grpc_service_pb2_grpc

    url = 'localhost:8001'
    model_name = 'tiny_averager'
    model_version = 1
    input_name = 'JpegBatch'
    output_name = 'Activation'

    channel = grpc.insecure_channel(url)
    grpc_stub = grpc_service_pb2_grpc.GRPCServiceStub(channel)

    print("Model status")
    request = grpc_service_pb2.StatusRequest(model_name=model_name)
    response = grpc_stub.Status(request)

    print(response)

    print("Inference using grpc")
    input_bytes = open('nr5.jpg', 'rb').read()
    input_size = len(input_bytes)

    request = grpc_service_pb2.InferRequest()
    request.meta_data.batch_size = 1

    request.meta_data.input.add(name=input_name, dims=[1], batch_byte_size=input_size)
    request.raw_input.append(input_bytes)

    request.model_name = model_name
    request.model_version = model_version
    request.meta_data.output.add(name="Activation")

    response = grpc_stub.Infer(request)
    print(response)
```

**Expected behavior**
Network should return an average color value of the image

**Comment by GuanLuo** [comment]
What is the error message that you received? Besides that, to send tensor with string data type to Triton, each element will need to be prepended with 4 bytes to represent the length of the string, for instance, if you have a string tensor with shape [1], then the raw data representation will be: length of element 1 + raw data of element 1

**Comment by deadeyegoodwin** [comment]
@baracoder You may also be interested in trying the new V2 clients that should make it easier to work with strings and grpc. You can read about them in the Roadmap section of the README. They will be released in 20.05 in a few weeks so if you want to try them now you will have to build both server and clients from the master branch. Here is a grpc example that shows it: https://github.com/NVIDIA/triton-inference-server/blob/c3/src/clients/python/experimental_api_v2/examples/grpc_v2_explicit_byte_content_client.py

**Comment by baracoder** [comment]
Thank you for the replies. Prefixing the actual input bytes with a length, solved the problem.

For reference:
```
    ...
    input_bytes = input_size.to_bytes(4, byteorder='little') + input_bytes
    input_size = input_size + 4
    ...
```
